contracts/fast/FastTokenFacet.sol

```diff
@@ -414,66 +368,71 @@ contract FastTokenFacet is AFastFacet, IERC20 {
    * - Emits a `Transfer(from, to, amount)` event.
    */
   function performTransfer(TransferArgs calldata p)
-      external onlyDiamondFacet
-      differentAddresses(p.from, p.to)
-      onlyTokenHolder(p.from)
-      onlyMarketplaceActiveMember(p.from)
-      onlyTokenHolder(p.to) {
+      external onlyDiamondFacet {
     // TODO: Make this function return instead of raising errors.
     // TODO: Make this function run even when a zero amount is passed. It should just emit.
-    LibFastToken.Data storage s = LibFastToken.data();
 
-    // Make sure that there's enough funds.
-    if (p.amount == 0) {
+    // Grab a pointer to our top-level storage.
+    LibFast.Data storage topData = LibFast.data();
+
+    // Requires that transfers are enabled for this FAST.
+    if (FastTopFacet(address(this)).transfersDisabled())
+      revert ICustomErrors.RequiresTransfersEnabled();
+    // Requires that `from` and `to` are different addresses.
+    else if (p.from == p.to)
+      revert ICustomErrors.RequiresDifferentSenderAndRecipient(p.from);
+    // Requires that `from` membership is active in the marketplace.
+    else if (!IHasActiveMembers(topData.marketplace).isMemberActive(p.from))
+      revert ICustomErrors.RequiresMarketplaceActiveMember(p.from);
+    // If `from` is not the reserve, requires that `from` is a valid token holder.
+    else if (p.from != address(0) && !canHoldTokens(p.from))
+      revert ICustomErrors.RequiresValidTokenHolder(p.from);
+    // If `from` is the reserve, requires that the message sender is an issuer member.
```

Comment:
Do we want to reffer to `msg.sender` or `spender` here given the context? (I know that `msg.sender` == `spender` at this point).